### PR TITLE
add redemption callback unittest

### PR DESCRIPTION
**Pull request by asalzmann**
## Context and purpose of the change

Add unittests to `RedemptionCallback`

## Brief Changelog

- Add success case test and failure case tests for `RedemptionCallback`
- NOTE: I did not test the `AddHostZoneToEpochUnbondingRecord` error; that function only errors when an epoch unbonding record is not found, but we verify the existence of the functions arguments a few lines above

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [X] tests

### CODE REVIEW

File: x/stakeibc/keeper/icacallbacks_redemption_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	s.App.RecordsKeeper.SetUserRedemptionRecord(s.Ctx(), userRedemptionRecord1)
+	s.App.RecordsKeeper.SetUserRedemptionRecord(s.Ctx(), userRedemptionRecord2)
+
+	packet := channeltypes.Packet{}
+	var msgs []sdk.Msg
+	msgs = append(msgs, &banktypes.MsgSend{})
+	ack := s.ICAPacketAcknowledgement(msgs)
+	callbackArgs := types.RedemptionCallback{
+		HostZoneId:            chainId,
+		UnbondingEpochNumbers: []uint64{epochNumber},
```
Comment (sampocs) [inline x/stakeibc/keeper/icacallbacks_redemption_test.go:78]: nit: should we update this proto field to `EpochUnbondingRecordIds` to align with undelegate?

File: x/stakeibc/keeper/icacallbacks_redemption_test.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
+package keeper_test
```
Comment (sampocs) [inline x/stakeibc/keeper/icacallbacks_redemption_test.go:1]: Just noticed this [typo](https://github.com/Stride-Labs/stride/blob/main/x/stakeibc/keeper/icacallbacks_redemption.go#L47). Should be "RedemptionCallback". Wanna fix it in this PR?

**Review by sampocs — APPROVED** [review]
Squeaky clean, good stuff!

File: x/stakeibc/keeper/icacallbacks_redemption_test.go (hunk 2 above)
Comment (asalzmann) [reply x/stakeibc/keeper/icacallbacks_redemption_test.go:1]: Where's the typo? Sorry, I think I'm missing it

File: x/stakeibc/keeper/icacallbacks_redemption_test.go (hunk 1 above)
Comment (asalzmann) [reply x/stakeibc/keeper/icacallbacks_redemption_test.go]: done

**Pull request merged**